ui/apps/dashboard/src/components/Environments/environment-context.tsx

```diff
@@ -0,0 +1,26 @@
+'use client';
+
+import { createContext, useContext } from 'react';
+
+import type { Environment } from '@/utils/environments';
+
+export const EnvironmentContext = createContext<Environment | undefined>(undefined);
+
+type EnvironmentProviderProps = {
+  env: Environment;
+  children: React.ReactNode;
+};
+
+export function EnvironmentProvider({ env, children }: EnvironmentProviderProps) {
```

Comment:
The env context provider no longer fetches the data itself on client render, we pass it in server side.